### ISSUE TITLE
Create Documentation How to Set Up Haveno for Development in Eclipse

### ISSUE THREAD

**Issue by eum2o** [haveno-dex/haveno#118]
It'd be helpful if there was some guidance how to set up this repository in Eclipse. In particular:

* How to configure Eclipse to use the wrapped Gradle or alternatively how to configure Eclipse's Gradle wrapper (so that sources compile eventually)?
* What project natures to configure?
* How to start Haveno from Eclipse (to be able to debug etc.)?

Since the .gitignore contains `.project`, I assume that developers are using Eclipse, too. Maybe it's also an option to commit such configuration (I also can imagine committing a physical Eclipse run configuration).

Creating such guide would ease the life of new potential developers/contributors.

**Comment by woodser** [comment]
See instructions to [import Haveno into Eclipse](https://github.com/haveno-dex/haveno/blob/master/docs/import-haveno.md#importing-haveno-into-eclipse-ide).

Feedback welcome.

**Comment by eum2o** [comment]
Thank you! This helps a lot.

> Feedback welcome.

I followed the `Importing Haveno into Eclipse IDE` steps and it basically worked fine (except some compilation problems, see "Compilation Problems" further below):
![image](https://user-images.githubusercontent.com/1593923/139735478-e88b92d7-dff4-4cd4-a112-5e7547825a48.png)

### Some minor issues
* The link on https://github.com/haveno-dex/haveno/blob/master/docs/README.md to "haveno-import.md" is broken (links to not existing https://github.com/haveno-dex/haveno/blob/master/docs/idea-import.md)
* https://github.com/haveno-dex/haveno/blob/master/docs/import-haveno.md#importing-haveno-into-eclipse-ide: 
  * references the image https://github.com/haveno-dex/haveno/blob/master/docs/edit_configurations.png which doesn't exist
  * The link to "dev-setup.md" in step 8 of the IntelliJ IDEA is broken
* I'd think about upgrading the default heap space size in the `eclipse.ini` file (e.g. to `-Xms2G` and `-Xmx8G`)

### Compilation Problems
* I have errors of which most seem to be compilation errors. 
  * Here are some examples of types that cannot be found: `org.junit.*`, `lombok.*` (my Eclipse knows Lombok but the project's classpath doesn't)
  * Full list: [errors.txt](https://github.com/haveno-dex/haveno/files/7455941/errors.txt)

This might be a Gradle configuration problem. I am not used to Gradle so I don't immediately see it. I just executed some Gradle tasks (in "build" & "build setup" and I also tried running `./gradlew clean` and rebuilt via _Project_ > _Clean..._ ) which didn't help. But I will have a closer look later. I wonder if I should configure Gradle properly (e.g. use a particular version).
![image](https://user-images.githubusercontent.com/1593923/139737550-1b745f1f-a8e8-4e78-ab40-e6969d52c43a.png)

#### Environment
* openjdk version "11.0.10" 2021-01-19
* Eclipse 2021-09 (4.21.0)
* Buildship: Eclipse Plug-ins for Gradle 3.1.5.v20210113-0929


I am thankful for hints about how to fix the compilation problems. I'll get back when I found out something myself.

**Comment by erciccione** [comment]
> The link on https://github.com/haveno-dex/haveno/blob/master/docs/README.md to "haveno-import.md" is broken (links to not existing https://github.com/haveno-dex/haveno/blob/master/docs/idea-import.md)

Nice catch. Fixing it right now (#153). The other problems you are pointing out come from the inteeliJ section, which is outdated. Maybe could be better to remove it entirely, but people using intelliJ could still find it useful, even if only partially.

**Comment by woodser** [comment]
> I have errors of which most seem to be compilation errors.

Do the errors persist after right clicking the project > Gradle > Refresh Gradle Project?

**Comment by eum2o** [comment]
> Do the errors persist after right clicking the project > Gradle > Refresh Gradle Project?

This led to a `org.gradle.api.InvalidUserDataException: Checksum failed for com.github.JesusMcCloud:jtorctl:6465e0` which I was able to fix with `./gradlew --info --refresh-dependencies` and it seems that the compilation problems (junit, lombok, etc.) mentioned above are resolved. I assume that I had some Gradle caching issues which prevented a successful build in the first place. I should have checked the logs earlier. Thanks for the hint!

The only compilation problems left were related to missing proto sources. I had a lot of errors like `bisq.proto cannot be resolved to a type`, `protobuf.OfferPayload`, etc. Apparently they're expected to be in `proto/build/generated/source/proto/main` which didn't exist. I manually executed the Gradle task _:proto > build > build_ which generated the missing proto sources. 

Now all problems are fixed. 

Is manually generating the proto sources something that needs to be put on the checklist? Due to my Gradle-noobishness I cannot judge if the `com.google.protobuf` plugin should have been executed automatically.

Thanks for your support.

**Comment by woodser** [comment]
Yeah there was a recent problem with jtorctl which requires refreshing dependencies, which `./gradlew --refresh-dependencies` fixes.

I've added a step to the Eclipse instructions to right click Eclipse > Gradle > Refresh Gradle Project. See https://github.com/haveno-dex/haveno/pull/156

`make` should generate the protobuf definitions. This step is documented in the instructions to install Haveno before importing to Eclipse. I've also made it clearer this should be done before importing to Eclipse.
